third_party/amd/lib/TritonAMDGPUTransforms/CanonicalizePointers.cpp

```diff
@@ -753,12 +753,21 @@ LogicalResult PointerCanonicalizer::rewriteSelectOp(arith::SelectOp selectOp,
   Value cond = selectOp.getCondition();
   // If we didn't traverse both operands, simply materialize the pointer
   if (!pointers.contains(trueVal) || !pointers.contains(falseVal))
-    return (materializeFatPointer(selectOp, curLoc, curOperand->get()));
+    return materializeFatPointer(selectOp, curLoc, curOperand->get());
 
   // If both have been traversed, then we can rewrite select of pointers as a
   // select of base and offset
   FatPtr fatPtrT = pointers[trueVal];
   FatPtr fatPtrF = pointers[falseVal];
+  nextPtr = selectOp.getResult();
+
+  // Simple case of a scalar select: update the base pointer
+  if (!isa<RankedTensorType>(selectOp.getType())) {
+    FatPtr fatPtr = pointers[trueVal];
+    pointers[nextPtr] = fatPtr.copyWithOffset(nextPtr);
+    nextPtr = selectOp.getResult();
```

Comment:
L768 is dead code?

Reply:
No, we need to set the `nextPtr` so that we can proceed the walk.

Reply:
Ah right. forgot that's a reference value.. I'm sort of still used to Google style guide regarding this--you'd use a pointer for such cases. But in LLVM/MLIR references are quite common.

Reply:
Actually you have it set at L762 already right?

Reply:
Actually you are right :) Removed L762